File: frontend/lib/models/persona.dart
class Persona {
  final String id;
  final String name;
  final String role;
  final String? greeting;
  final String? avatarEmoji;
  final String? personality;
  final String? systemPrompt;
  final String? voice;
  final bool active;
  final String subscriptionTier;
  final String? userId;

  Persona({
    required this.id,
    required this.name,
    required this.role,
    this.greeting,
    this.avatarEmoji,
    this.personality,
    this.systemPrompt,
    this.voice,
    required this.active,
    required this.subscriptionTier,
    this.userId,
  });

  bool get isOwned => userId != null;

  factory Persona.fromJson(Map<String, dynamic> json) {
    return Persona(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      role: json['role'] ?? '',
      greeting: json['greeting'],
      avatarEmoji: json['avatarEmoji'],
      personality: json['personality'],
      systemPrompt: json['systemPrompt'],
      voice: json['voice'],
      active: json['active'] ?? true,
      subscriptionTier: json['subscriptionTier'] ?? 'free',
      userId: json['userId'],
    );
  }
}

File: backend/src/main/java/com/innercircle/model/Persona.java
package com.innercircle.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "personas")
@Data
@NoArgsConstructor
public class Persona {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String name;
    private String role;
    private String avatarEmoji;
    private String personality;
    private String voice;

    @Column(columnDefinition = "TEXT")
    private String systemPrompt;

    private String greeting;

    @Column(name = "nsfw_enabled")
    private boolean nsfwEnabled = false;

    @Enumerated(EnumType.STRING)
    private SubscriptionTier subscriptionTier = SubscriptionTier.free;

    @Column(name = "is_active")
    private boolean active = true;

    @ManyToOne
    @JoinColumn(name = "owner_user_id")
    @com.fasterxml.jackson.annotation.JsonIgnore
    private User owner;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @com.fasterxml.jackson.annotation.JsonIgnore
    private User user;

    @CreationTimestamp
    private Instant createdAt;
}
